### PR TITLE
UFS-dev PR#92

### PR DESCRIPTION
**Pull request by grantfirl**
Identical to https://github.com/ufs-community/ufs-weather-model/pull/1844

Contains changes from #107 until it is merged.

### CODE REVIEW

**Comment by grantfirl** [comment]
Expected BL changes:
COMPILE | rrfs | intel | -DAPP=ATM -DCCPP_SUITES=FV3_RAP,FV3_RAP_sfcdiff,FV3_HRRR,FV3_RRFS_v1beta,FV3_RRFS_v1nssl -D32BIT=ON | | fv3 |
RUN | hrrr_control                                      | - wcoss2                   | baseline |
RUN | hrrr_control_qr                                   | - wcoss2                   |          |
RUN | hrrr_control_decomp                               | - wcoss2                   |          |
RUN | hrrr_control_2threads                             | - wcoss2                   |          |
RUN | hrrr_control_restart                              | - wcoss2                   |          | hrrr_control
RUN | hrrr_control_restart_qr                           | - wcoss2                   |          | hrrr_control_qr

COMPILE | atm_debug_dyn32 | intel | -DAPP=ATM -DDEBUG=ON -D32BIT=ON -DCCPP_SUITES=FV3_HRRR,FV3_GFS_v16,FV3_GFS_v16_csawmg,FV3_GFS_v16_ras,FV3_GFS_v17_p8,FV3_GFS_v15_thompson_mynn_lam3km,FV3_RAP,FV3_RAP_unified_ugwp,FV3_RAP_cires_ugwp,FV3_RAP_flake,FV3_RAP_clm_lake,FV3_RAP_noah,FV3_RAP_sfcdiff,FV3_RAP_noah_sfcdiff_cires_ugwp,FV3_RRFS_v1beta | | fv3 |
RUN | hrrr_control_debug                                |                            | baseline |
RUN | rap_clm_lake_debug                                |                            | baseline |

COMPILE | rrfs_dyn32_phy32 | intel | -DAPP=ATM -DCCPP_SUITES=FV3_RAP,FV3_HRRR -D32BIT=ON -DCCPP_32BIT=ON | | fv3 |
RUN | hrrr_control_dyn32_phy32                          |                            | baseline |
RUN | hrrr_control_qr_dyn32_phy32                       |                            | baseline |
RUN | hrrr_control_2threads_dyn32_phy32                 |                            |          |
RUN | hrrr_control_decomp_dyn32_phy32                   |                            |          |
RUN | hrrr_control_restart_dyn32_phy32                  |                            |          | hrrr_control_dyn32_phy32
RUN | hrrr_control_restart_qr_dyn32_phy32               |                            |          | hrrr_control_qr_dyn32_phy32

COMPILE | rrfs_dyn32_phy32_debug | intel | -DAPP=ATM -DCCPP_SUITES=FV3_RAP,FV3_HRRR -D32BIT=ON -DCCPP_32BIT=ON -DDEBUG=ON | | fv3 |
RUN | hrrr_control_debug_dyn32_phy32                    |                            | baseline |

COMPILE | rrfs | gnu | -DAPP=ATM -DCCPP_SUITES=FV3_RAP,FV3_RAP_sfcdiff,FV3_HRRR,FV3_RRFS_v1beta -D32BIT=ON | + hera cheyenne | fv3 |
RUN | hrrr_control                                      | + hera cheyenne            | baseline |
RUN | hrrr_control_qr                                   | + hera cheyenne            |          |
RUN | hrrr_control_2threads                             | + hera cheyenne            |          |
RUN | hrrr_control_decomp                               | + hera cheyenne            |          |
RUN | hrrr_control_restart                              | + hera cheyenne            |          | hrrr_control
RUN | hrrr_control_restart_qr                           | + hera cheyenne            |          | hrrr_control_qr

COMPILE | atm_dyn32_debug | gnu | -DAPP=ATM -D32BIT=ON -DDEBUG=ON | + hera cheyenne | fv3 |
RUN | hrrr_control_debug                                | + hera cheyenne            | baseline |
RUN | rap_clm_lake_debug                                | + hera cheyenne            | baseline |

COMPILE | rrfs_dyn32_phy32 | gnu | -DAPP=ATM -DCCPP_SUITES=FV3_RAP,FV3_HRRR -D32BIT=ON -DCCPP_32BIT=ON | + hera cheyenne | fv3 |
RUN | hrrr_control_dyn32_phy32                          | + hera cheyenne            | baseline |
RUN | hrrr_control_qr_dyn32_phy32                       | + hera cheyenne            | baseline |
RUN | hrrr_control_2threads_dyn32_phy32                 | + hera cheyenne            |          |
RUN | hrrr_control_decomp_dyn32_phy32                   | + hera cheyenne            |          |
RUN | hrrr_control_restart_dyn32_phy32                  | + hera cheyenne            |          | hrrr_control_dyn32_phy32
RUN | hrrr_control_restart_qr_dyn32_phy32               | + hera cheyenne            |          | hrrr_control_qr_dyn32_phy32

COMPILE | atm_dyn32_phy32_debug | gnu | -DAPP=ATM -D32BIT=ON -DCCPP_32BIT=ON -DDEBUG=ON | + hera cheyenne | fv3 |
RUN | hrrr_control_debug_dyn32_phy32                    | + hera cheyenne            | baseline |

**Comment by mkavulich** [comment]
@grantfirl Can you remove the old tests under `/scratch1/BMC/gmtb/CCPP_regression_testing/NCAR_ufs-weather-model/run_gjf` when you get a chance? I just want to make sure we don't run into disk space issues while flying through these tests.

**Comment by grantfirl** [comment]
I removed everything in the directory, but I don't have permissions to
delete the directory itself. That's weird.

**Comment by grantfirl** [comment]
@mkavulich This one is ready to start testing to verify failed tests.

**Comment by grantfirl** [comment]
@mkavulich Don't we need to do hera-intel-RT first to verify failed tests?

**Comment by mkavulich** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: intel
Job: BL
[BL] Repo location: /scratch1/BMC/gmtb/CCPP_regression_testing/NCAR_ufs-weather-model//run//1584956840/20240111214517/ufs-weather-model
Please make changes and add the following label back: hera-intel-BL

**Comment by grantfirl** [comment]
@mkavulich It looks like the baselines were created successfully, and we can use those, but we still need to run RTs against the old baseline.

**Comment by mkavulich** [comment]
Yeah @grantfirl sorry about that, I figured it didn't matter the order in which we did this so long as there were no unexpected failures.

I'm still trying to remember/discover the exact process that's happening in the "new baseline" tests...my initial thoughts would be that it creates a new baseline and runs again against to compare against that same baseline, does that sound correct? It's not actually clear from the logs that that's what's happening, but I haven't had a chance to delve deeply in there yet (they are *exceedingly* hard to parse)

**Comment by mkavulich** [comment]
I'm also not sure why the BL tests are showing a failure. It seems to be some check that is running within the auto-RT on the repository side (rather than the copy staged on disk) that I don't understand. I will keep investigating.

**Comment by grantfirl** [comment]
> Yeah @grantfirl sorry about that, I figured it didn't matter the order in which we did this so long as there were no unexpected failures.
> 
> I'm still trying to remember/discover the exact process that's happening in the "new baseline" tests...my initial thoughts would be that it creates a new baseline and runs again against to compare against that same baseline, does that sound correct? It's not actually clear from the logs that that's what's happening, but I haven't had a chance to delve deeply in there yet (they are _exceedingly_ hard to parse)

I guess that the order of RT/BL isn't too important as long as there aren't any unexpected failures in RT, which is why it has typically gone first. You don't want to create new baselines until you are sure that the code is performing as expected.

I also found the scripts super confusing and Dustin and I had a similar discussion about whether RTs are run to check against new baselines (basically just testing reproducibility). I think that we came to the conclusion that it _was_ doing another RT after the baseline was recreated, but I don't think that we were ever 100% certain of this. Since the UFS/EMC/EPIC code managers are ok with only the RT/BL tags, we were assuming that this was good enough for us too.

**Comment by mkavulich** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: intel
Job: RT
[RT] Repo location: /scratch1/BMC/gmtb/CCPP_regression_testing/NCAR_ufs-weather-model//run//1584956840/20240112164518/ufs-weather-model
[RT] Error: Test 068 hrrr_control_intel FAIL Tries: 2
[RT] Error: Test 069 hrrr_control_qr_intel FAIL Tries: 2
[RT] Error: Test 070 hrrr_control_decomp_intel FAIL Tries: 2
[RT] Error: Test 071 hrrr_control_2threads_intel FAIL Tries: 2
[RT] Error: Test 104 hrrr_control_debug_intel FAIL Tries: 2
[RT] Error: Test 115 rap_clm_lake_debug_intel FAIL Tries: 2
[RT] Error: Test 120 hrrr_control_dyn32_phy32_intel FAIL Tries: 2
[RT] Error: Test 121 hrrr_control_qr_dyn32_phy32_intel FAIL Tries: 2
[RT] Error: Test 123 hrrr_control_2threads_dyn32_phy32_intel FAIL Tries: 2
[RT] Error: Test 124 hrrr_control_decomp_dyn32_phy32_intel FAIL Tries: 2
[RT] Error: Test 134 hrrr_control_debug_dyn32_phy32_intel FAIL Tries: 2
[RT] Error: Test 196 hrrr_control_gnu FAIL Tries: 2
[RT] Error: Test 197 hrrr_control_qr_gnu FAIL Tries: 2
[RT] Error: Test 198 hrrr_control_2threads_gnu FAIL Tries: 2
[RT] Error: Test 199 hrrr_control_decomp_gnu FAIL Tries: 2
[RT] Error: Test 213 hrrr_control_debug_gnu FAIL Tries: 2
[RT] Error: Test 225 rap_clm_lake_debug_gnu FAIL Tries: 2
[RT] Error: Test 228 hrrr_control_dyn32_phy32_gnu FAIL Tries: 2
[RT] Error: Test 229 hrrr_control_qr_dyn32_phy32_gnu FAIL Tries: 2
[RT] Error: Test 231 hrrr_control_2threads_dyn32_phy32_gnu FAIL Tries: 2
[RT] Error: Test 232 hrrr_control_decomp_dyn32_phy32_gnu FAIL Tries: 2
[RT] Error: Test 242 hrrr_control_debug_dyn32_phy32_gnu FAIL Tries: 2
[RT] Log file shows failures.
[RT] Please obtain logs from /scratch1/BMC/gmtb/CCPP_regression_testing/NCAR_ufs-weather-model//run//1584956840/20240112164518/ufs-weather-model

**Comment by mkavulich** [comment]
@grantfirl It looks like the `rap_clm_lake_debug` tests are unexpected failures, can you confirm that's the case? Otherwise everything else looks expected.

**Review by mkavulich — APPROVED** [review]
@grantfirl Sorry, forgot to approve this one 😬 

How did you want to go about updating the test logs and baseline location? I could go ahead and do it once tests are successful since I think I have write permissions on your fork, but I don't want to commit things to your branch if you're working on it.

**Comment by grantfirl** [comment]
> @grantfirl Sorry, forgot to approve this one 😬
> 
> How did you want to go about updating the test logs and baseline location? I could go ahead and do it once tests are successful since I think I have write permissions on your fork, but I don't want to commit things to your branch if you're working on it.

I can do that. I'll go ahead and start the merge process so that we can move on to the next one.

**Comment by grantfirl** [comment]
> /scratch1/BMC/gmtb/CCPP_regression_testing/NCAR_ufs-weather-model//run//1584956840/20240112164518/ufs-weather-model

@mkavulich I still can't mv the new baselines into the baselines directory on Hera due to lack of permissions. I've done the rest. If you could mv the new baselines and give it a name of main-20240116, we'll be ready to continue with the next one once I update the branches.
